packages/chevrotain/src/scan/reg_exp.ts

```diff
@@ -130,10 +132,10 @@ export function firstChar(ast): number[] {
 
                 // reached a mandatory production, no more start codes can be found on this alternative
                 if (
-                    //
-                    atom.quantifier === undefined ||
-                    (atom.quantifier !== undefined &&
-                        atom.quantifier.atLeast > 0)
+                    // for groups, we need to walk the tree to determine whether their "innards" are
+                    // wholly optional or not; for non-groups, a simple quantifier check works
+                    (atom.quantifier === undefined && atom.type !== "Group") ||
```

Comment:
I think there may be a bug here.
`quantifier === undefined` would be mandatory for a none group.
But also quantifier.atLeast > 0.

```javascript
/a?b{1,2}c/
```

So our start Chars would contain too many results and that could slow down the lexer.

Reply:
I'm not following quite yet. I ran that pattern through my changes and it produced `[97, 98]`, which looks to be the right result here. Do you mean that you think it should short circuit earlier?

Reply:
I will debug it, perhaps I am mistaken 😄

Reply:
BTW, just in case it's relevant (and may have been missed), note that there _is_ (implicitly) a `quantifier.atLeast > 0` check in `isWholeOptional` now: https://github.com/SAP/chevrotain/pull/934/files#diff-3359a0c7c17155c58a71d93f6d581608R186. So, if a (scalar) atom has a quantifier that is greater than 0 and it is passed to `isWholeOptional`, the base case of `!ast.value` will be met (on the second step through) and `isWholeOptional` will correctly return `false`.

Reply:
Have a look here:
https://github.com/SAP/chevrotain/commit/e47bd96719c4c4ff5f58d64eb344f268601c9cc2#diff-3359a0c7c17155c58a71d93f6d581608

I tried to make the condition less implicit, if this looks good I can release a new version 😄

Reply:
That looks good to me!

Reply:
Try version 4.3.2 it is release on npm now.

Reply:
@bd82 I tried 4.3.2 and all worked great! 😄